src/controllers/user/routes.ts

```diff
@@ -1,12 +1,18 @@
 import { Router } from 'express';
 import UserController from './Controller';
-import validation from './validation';
+import validation from '../user/validation';
+import { authMiddleware } from '../../libs/routes';
 import validationHandler from '../../libs/validationHandler';
 const userRouter: Router = Router();
 
 userRouter.route('/')
-.get(validationHandler( validation.get), UserController.get)
-.post( validationHandler( validation.create ), UserController.create)
-.put(validationHandler(validation.update ), UserController.update);
+.get(authMiddleware('getUser', 'read'), validationHandler( validation.get), UserController.get)
+.post(authMiddleware('getUser', 'read'), validationHandler( validation.create ), UserController.create)
+.put(authMiddleware('getUser', 'read'), validationHandler(validation.update ), UserController.update);
  userRouter.route('/:id').delete(validationHandler( validation.delete ), UserController.delete);
+
+ userRouter.route('/me')
+ .get(authMiddleware('getUser', 'read'), validationHandler( validation.get), UserController.me);
+ userRouter.route('/login')
+ .post(authMiddleware('getUser', 'read'), validationHandler(validation.login), UserController.login);
```

Comment:
Do not put authMiddleware on the "/login" route. Else it will not allow to login